### ISSUE TITLE
feature: demand perfect accuracy to unlock a key

### ISSUE THREAD

**Issue by aradzie** [aradzie/keybr.com#34]
In addition to the target typing speed, add a requirement of perfect accuracy to unlock a key.

**Comment by Krishnakantpore** [comment]
Yeah, I'd say the previous confidance level menthol was better for me.

**Comment by aradzie** [comment]
> Yeah, I'd say the previous confidence level menthol was better for me.

It's still there. The only difference is that now the confidence level is computed against the custom speed threshold. Previously it was a constant of 35WPM.

**Comment by Kevin-exe** [comment]
I'm gonna add some of my ideas to this feature thread too:

It would be really convenient to have an "accuracy training mode" where the lesson restarts if you make too many mistakes (with a number determined by the user, like 10, 3, or even 1 mistake to force 100% accuracy) I know you can hit the escape key to restart but it could open a new door to allow judging user metrics and key unlock progression more directly biased towards accuracy instead of speed.
This would be useful in learning a new keyboard layout, where I so often hit the wrong key thinking of QWERTY while on Colemak, and it feels I am more rewarded to type faster with many mistakes than slower with few mistakes. And not being punished for making a mistake.

**Comment by Olmo-Gutierrez** [comment]
Hey just an addition, no need for perfect accuracy, I would preferably do a slider with the % you would like

**Comment by Krishnakantpore** [comment]
> > Yeah, I'd say the previous confidence level menthol was better for me.
> 
> It's still there. The only difference is that now the confidence level is computed against the custom speed threshold. Previously it was a constant of 35WPM.

I think there is something off here, because even if I set my Target typing speed is 40 WPM, I was still able to complete it even if my accuracy was 90%.

Another thing is, if I can recall correctly,
as I unlock more letters, the software would take me back to previous letters where my Accuracy dropped significantly. That doesn't happen now.
